actionview/lib/action_view/base.rb

```diff
@@ -197,17 +198,49 @@ def assign(new_assigns) # :nodoc:
       @_assigns = new_assigns.each { |key, value| instance_variable_set("@#{key}", value) }
     end
 
-    def initialize(context = nil, assigns = {}, controller = nil, formats = nil) #:nodoc:
+    # :stopdoc:
+
+    def self.build_renderer(context, controller, formats)
+      lookup_context = context.is_a?(ActionView::LookupContext) ?
+        context : ActionView::LookupContext.new(context)
+      lookup_context.formats  = formats if formats
+      lookup_context.prefixes = controller._prefixes if controller
+      ActionView::Renderer.new(lookup_context)
+    end
+
+    def self.empty
+      with_view_paths([])
+    end
+
+    def self.with_view_paths(view_paths, assigns = {}, controller = nil)
+      with_context ActionView::LookupContext.new(view_paths), assigns, controller
+    end
+
+    def self.with_context(context, assigns = {}, controller = nil)
+      new ActionView::Renderer.new(context), assigns, controller
+    end
+
+    NULL = Object.new
+
+    # :startdoc:
+
+    def initialize(renderer, assigns = {}, controller = nil, formats = NULL) #:nodoc:
       @_config = ActiveSupport::InheritableOptions.new
 
-      if context.is_a?(ActionView::Renderer)
-        @view_renderer = context
+      unless formats == NULL
```

Comment:
If formats is `NULL`, should change the value to `nil`? 
If this is left as `NULL(Object.new)`, it will be incorrectly judged where the value is nil or not. 

For example, if using web-console, `NoMethodError` will be raise in lookup context. 

```
DEPRECATION WARNING: ActionView::Base instances should be constructed with a view renderer,
assigments, and a controller.
 (called from new at /home/y-yagi/.rbenv/versions/2.6.0/lib/ruby/gems/2.6.0/gems/web-console-3.7.0/lib/web_console/template.rb:21)

NoMethodError: undefined method `delete' for #<Object:0x000055e2e6120880>
	from /home/y-yagi/src/rails/master_y_yagi/rails/actionview/lib/action_view/lookup_context.rb:258:in `formats='
	from /home/y-yagi/src/rails/master_y_yagi/rails/actionview/lib/action_view/base.rb:206:in `build_renderer'
	from /home/y-yagi/src/rails/master_y_yagi/rails/actionview/lib/action_view/base.rb:243:in `initialize'
	from /home/y-yagi/.rbenv/versions/2.6.0/lib/ruby/gems/2.6.0/gems/web-console-3.7.0/lib/web_console/template.rb:21:in `new'
	from /home/y-yagi/.rbenv/versions/2.6.0/lib/ruby/gems/2.6.0/gems/web-console-3.7.0/lib/web_console/template.rb:21:in `render'
	from /home/y-yagi/.rbenv/versions/2.6.0/lib/ruby/gems/2.6.0/gems/web-console-3.7.0/lib/web_console/middleware.rb:37:in `block in call'
	from /home/y-yagi/.rbenv/versions/2.6.0/lib/ruby/gems/2.6.0/gems/web-console-3.7.0/lib/web_console/middleware.rb:20:in `catch'
	from /home/y-yagi/.rbenv/versions/2.6.0/lib/ruby/gems/2.6.0/gems/web-console-3.7.0/lib/web_console/middleware.rb:20:in `call'
	from /home/y-yagi/src/rails/master_y_yagi/rails/actionpack/lib/action_dispatch/middleware/show_exceptions.rb:33:in `call'
```

Reply:
@y-yagi I think so. Can you commit the change?

Reply:
Already fixed by eda0f574f129fcd5ad1fc58b55cb6d1db71ea95c.